installer.go

```diff
@@ -77,7 +77,12 @@ func (i *Installer) InstallDependency(dep Dependency, outputDir string) error {
 		return ExtractTarXz(tmpFile, outputDir)
 	}
 
-	return ExtractTarGz(tmpFile, outputDir)
+	if strings.HasSuffix(entry.URI, ".tar.gz") || strings.HasSuffix(entry.URI, ".tgz") {
```

Comment:
I'm not sure if we want to require URIs to have a `.tgz` or `.tar.gz` extension to be extracted. We should double-check that all uses of `uri` in the CF buildpacks follow that convention before merging (or consider a different mechanism for indicating the type of download).

Reply:
I'm not sure I follow sorry. Before this PR, it assumed that all urls that didn't get handled by previous `if` statements were `.tar.gz` and would try to unpack them. My PR puts this unpacking into a new `if` statement. The new default behavior after my PR is to *not* unpack the file.

Can you give an example of where this PR breaks something for a buildpack? I thought my PR was only adding new functionality; I wasn't able to imagine that it might break any existing buildpack.

Reply:
@sclevine let me know how you're thinking this PR should be changed and I'll investigate asap

Reply:
@drnic The addition of the check on the if block changes the logic around _when_ `ExtractTarGz` gets called to make it more specific, with `CopyFile` now being the fall-through/default case. @sclevine is worried that assets that aren't explicititely named as .tar.gz, etc. would have been treated as such by default (before this PR) and now won't. This is going to require us to do bit of legwork to track down all of our assets. Maybe if you have a specific use case you could add that clause? Otherwise I think we need a better way than altering the default case.